kafka-connect/build.gradle

```diff
@@ -1,5 +1,6 @@
 dependencies {
   implementation project(":iceberg-kafka-connect-events")
+  implementation project(":iceberg-kafka-connect-deadletter")
```

Comment:
Why are we depending on the dead-letter module here?

I see no reason to force this dependency; just like we don't force this project to depend on transforms subproject. 

Note: I would be fine with the inverse (dead-letter module **_can_** depend on kafka-connect module).

Reply:
You need to depend on it to see the interface for FailedRecordFactory